pkg/upgraders/osd/upgrader.go

```diff
@@ -142,11 +143,19 @@ func PreClusterHealthCheck(c client.Client, cfg *osdUpgradeConfig, scaler scaler
 
 	ok, err := performClusterHealthCheck(c, metricsClient, cvClient, cfg, logger)
 	if err != nil || !ok {
-		metricsClient.UpdateMetricClusterCheckFailed(upgradeConfig.Name)
+		upgradeConfig.Status.History[0].HealthCheck.Failed = true
```

Comment:
imo everything in upgraders should be a condition. (returned from the upgrader)
These statuses will change between upgraders so we can't define them on the CRD

Reply:
Actually, everything outside the controllers should be

Reply:
Yeah.. agree that this makes sense (upon reflection of the related comment about Notifications in the CRD).

We should ideally want upgrade steps to be flexible and adaptable and not necessarily have any guarantees of them needing to be present, having them explicitly modelled in the CRD goes against that somewhat.